src/individual_metrics.py

```diff
@@ -0,0 +1,118 @@
+
+# from pydriller.domain.commit import ModificationType
+
+
+def get_commit_average(lines, commits):
+    """Find average lines modified per commit."""
+    # Loop through the dictionary and calculate the average lines per commits
+    # formula for average: (added + deleted)/number_of_commits
+    if commits != 0:
+        average = lines / commits
+        return (int)(average)
+    return 0
+
+
+def parse_for_type(name):
+    """Parse through file name and returns its format."""
+    if "." in name:
+        dot_index = name.find(".")
+        return name[dot_index:]
+    return name
+
+
+def get_file_formats(files):
+    """Create a list of unique file formats."""
+    formats = []
+    for file in files:
+        current_format = parse_for_type(file)
+        if current_format not in formats:
+            formats.append(current_format)
+    return formats
+
+
+# NOTE: this function is temporary to test get_individual_metrics considering
+# there is not a main module that connects eerything yet
+def add_raw_data_to_json(path_to_repo, json_file_name):
+    """Check if raw data is in .json file and collects it if not."""
+    # get the data from .json file
+    current_data = json_handler.get_dict_from_json_file(json_file_name)
+    # check if data has not been collected and do so if it was not
+    if "RAW_DATA" not in current_data.keys():
+        print("Raw data has not been collected, collecting it now...")
+        # collects data from data_collection
+        raw_data = {"RAW_DATA": data_collection.collect_commits_hash(path_to_repo)}
+        # Write raw data to .json file
+        json_handler.add_entry(raw_data, json_file_name)
+        print("Data collected")
+
+
+def calculate_individual_metrics(json_file_name):
+    """Retrieve the data from .json file and create a dictionary keyed by user."""
+    current_data = json_handler.get_dict_from_json_file(json_file_name)
+    # creates a hashmap where the key is the authors username
+    data_list = {}
+    # Check if RAW_DATA is in json
+    if "RAW_DATA" in current_data.keys():
+        for key in current_data["RAW_DATA"]:
+            author = key["author_name"]
+            email = key["author_email"]
+            # NOTE check date compatibility with json
+            # date = "N/A"
+            # check if the key already in in the dicitionary
+            if author in data_list:
+                # condition passed, adds one to the number of commits
+                data_list[author]["COMMITS"] += 1
+            else:
+                # condition fails, creates a new key and adds empty data
+                data_list[author] = {
+                    "EMAIL": email,
+                    "COMMITS": 1,
+                    "ADDED": 0,
+                    "REMOVED": 0,
+                    "TOTAL": 0,
+                    "MODIFIED": 0,
+                    "RATIO": 0,
+                    "FILES": [],
+                    "FORMAT": [],
+                    "COMMITDATE": [],
```

Comment:
Is this used throughout the rest of the code? I think that getting the commit date is important, but I know that this provides a confusing list of dates, that would have to be iterated through. I believe that this aspect of the individual metrics features can be removed.

Reply:
While the Ratio, files, and format are used, the `commitdate` is not yet. This is due to an issue with storing dates in the .json file. I plan on working with the data collection team to resolve this issue. I also agree that it would not be ideal to have it in individual metrics as of right now, I think that we might find a use for it later.

Reply:
I made the necessary changes to remove the commit date metric with my latest commits, please let me know if there are other metrics that you think we should add in the future.